packages/block-library/src/site-title/edit.js

```diff
@@ -36,4 +39,19 @@ export default function SiteTitleEdit() {
 			/>
 		</>
 	);
+
+	const viewMode = (
+		<h1>{ title }</h1>
+	);
+
+	return ( props.canUpdate && props.isSelected ) ? editMode : viewMode;
 }
+
+export default compose( [
+	withSelect( ( select, {} ) => {
+		const { canUser } = select( 'core' );
+		return {
+			canUpdate: canUser( 'update', 'settings' ),
```

Comment:
Just a note that I intend to make this part of the framework.

I envision that `core-data` will keep track of which entity properties are locked, actions can change that, entity configs provide default values/selectors, and `useEntityProp` will return a third boolean variable that says whether the specified prop is locked or not.